fix: remove unnecessary title attributes from datagrid

davidmenendez opened the pull request:
Closes #5351 

this PR removes a bunch of unnecessary `title` attributes that were strung throughout the `Datagrid` component that was causing accessibility issues. [current documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns) tells us there's only a few valid instances in which to use `title` and none of the instances here meet the criteria.

> Outside of potential uses on input elements, there are notable instances where the title is the recommended attribute for accessibility user experiences.

matthewgallo commented on packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx:
```diff
@@ -237,7 +236,6 @@ export const DatagridContent = ({
           [`${blockClass}__with-pagination`]: DatagridPagination,
           [`${blockClass}__dense-header`]: useDenseHeader,
         })}
-        title={gridTitle}
```
This prop isn't a `title` attribute but an actual prop for the `TableContainer` from `@carbon/react` which displays an optional title for the table.
![Screenshot 2024-06-24 at 3 46 00 PM](https://github.com/carbon-design-system/ibm-products/assets/10215203/e9d1fbf6-c3a3-43a5-b7dd-def700905ae3)

davidmenendez replied:
oooh i see. ok. my mistake. will revert those changes.

davidmenendez commented:
@matthewgallo thanks for catching that title issue. i went ahead and reverted that commit and only removed instances related to `title`.

i also added an additional small fix that adds `role: undefined` to cells and rows to alleviate violations that were showing up in the accessibility checker while testing

<img width="861" alt="Screenshot 2024-06-24 at 3 26 42 PM" src="https://github.com/carbon-design-system/ibm-products/assets/6370760/18d48f90-8589-4a32-9acd-a1b150f48f70">

took us down from 209 to 59 violations. progress 😅 

<img width="368" alt="Screenshot 2024-06-24 at 3 25 14 PM" src="https://github.com/carbon-design-system/ibm-products/assets/6370760/b4a40de4-b362-4c70-86a2-4b0d516332c3">
<img width="387" alt="Screenshot 2024-06-24 at 3 25 19 PM" src="https://github.com/carbon-design-system/ibm-products/assets/6370760/2e1eda48-4026-4b48-a534-90afd3132e73">

davidmenendez commented:
the last commit i added removed some additional problematic roles. this gets our violation count down to 32 💪

davidmenendez commented:
actually went ahead and addressed #4032 in the last commit because the fix was basically the same
